packages/ai-code-completion/src/browser/code-completion-postprocessor.ts

```diff
@@ -0,0 +1,48 @@
+// *****************************************************************************
+// Copyright (C) 2024 EclipseSource GmbH.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0
+// *****************************************************************************
+
+import { inject, injectable } from '@theia/core/shared/inversify';
+import { PreferenceService } from '@theia/core/lib/browser';
+import { PREF_AI_INLINE_COMPLETION_STRIP_BACKTICKS } from './ai-code-completion-preference';
+
+export interface CodeCompletionPostProcessor {
+    postProcess(text: string): string;
+}
+export const CodeCompletionPostProcessor = Symbol('CodeCompletionPostProcessor');
+
+@injectable()
+export class DefaultCodeCompletionPostProcessor {
+
+    @inject(PreferenceService)
+    protected readonly preferenceService: PreferenceService;
+
+    public postProcess(text: string): string {
+        if (this.preferenceService.get<boolean>(PREF_AI_INLINE_COMPLETION_STRIP_BACKTICKS, true)) {
+            return this.stripBackticks(text);
+        }
+        return text;
+    }
+
+    public stripBackticks(text: string): string {
+        if (text.startsWith('```')) {
+            // Remove the first backticks and any language identifier
+            const startRemoved = text.slice(3).replace(/^\w*\n/, '');
+            const secondBacktickIndex = startRemoved.indexOf('```');
```

Comment:
shouldn't we use lastIndexOf ?
This would better catch this case:
` ```\nFoo```Bar```FooBar``` `
I would expect the result to be: ` Foo```Bar```FooBar `
but in your code I assume it would be: `Foo`